common/src/main/java/org/apache/celeborn/common/network/server/memory/ReadBufferDispatcher.java

```diff
@@ -48,7 +48,7 @@ public void addBufferRequest(ReadBufferRequest request) {
 
   public void recycle(ByteBuf buf) {
     int bufferSize = buf.capacity();
-    buf.release();
+    buf.release(buf.refCnt());
     memoryManager.changeReadBufferCounter(-1 * bufferSize);
```

Comment:
We can delete buf.retain() when allocators new buffer as we always retain it when we use the buffer.